styleguide-csharp.md

```diff
@@ -0,0 +1,116 @@
+# Explore Education Statistics - C# Style Guide
+
+<details markdown="1">
+  <summary>Table of Contents</summary>
+
+-   [1 Background](#s1-background)
+-   [2 C# Language Rules](#s2-csharp-language-rules)
+    *   [2.1 Style](#s2.1-style)
+        +   [2.1.1 Named Parameters](#s2.1.1-named-parameters)
+    *   [2.2 Structure](#s2.2-structure)
+        +   [2.2.1 Entity Framework (EF)](#s2.2.1-ef)
+            +   [2.2.1.1 Configuration](#s2.2.1.1-ef-configuration)
+                +   [2.2.1.1.1 Annotations VS Fluent API](#s2.2.1.1.1-annotation-vs-fluent-api)
+            +   [2.2.1.2 Stored Procedures VS Linq VS SQL](#s2.2.1.2-stored-procedures-vs-linq-vs-sql)
+    *   [2.3 Best Practices](#s2.3-best-practices)
+        +   [2.3.2 Entity Framework (EF)](#s2.3.2-ef)
+-   [3 C# Testing Rules](#s3-csharp-testing-rules)
+    *   [3.1 Style](#s3.1-style)
+        +   [3.1.1 Test Naming Conventon](#s3.1.1-test-naming-convention)
+    *   [3.2 Test Setup](#s3.2-test-setup)
+    *   [3.3 Best Practices](#s3.3-best-practices)
+        +   [3.3.1 Synchronous VS Asynchronous Test Methods](#s3.3.1-synchronous-vs-asynchronous-test-methods)
+
+</details>
+
+<a id="s1-background"></a>
+## 1 Background 
+
+C# is the backend language used in this repository. This style guide is a list
+of agreed coding standards to conform to when writing C# code for this repository.
+
+
+<a id="s2-csharp-language-rules"></a>
+## 2 C# Language Rules 
+
+<a id="s2.1-style"></a>
+### 2.1 Style 
+
+<a id="s2.1.1-named-parameters"></a>
+### 2.1.1 Named Parameters
+
+- As a rule, if we hit 3 or more parameters for a method call, consider using a named parameters, or break out a class/record to hold the parameters*
+- If we have multiple parameters of the same type, use named parameters to protect against the accidental reordering or parameters*
```

Comment:
` reordering or parameters` should be ` reordering of parameters` here

Reply:
good spot!